moldesign/_tests/test_atom_containers.py

```diff
@@ -129,10 +136,86 @@ def test_position_links(fixturename, request):
     np.testing.assert_array_equal(obj.positions[0, :],
                                   obj.atoms[0].position)
 
+
 def _get_minimum_pairwise(group1, group2):
     mindist = np.inf * u.angstrom
     for a1, a2 in itertools.product(group1.atoms, group2.atoms):
         distance = (a1.position-a2.position).norm()
         mindist = min(distance, mindist)
 
     return mindist
+
+
+@pytest.mark.parametrize('fixturename', ['atom', 'residue', 'atomlist', 'small_molecule'])
+def test_atoms_within(fixturename, request):
+    obj = request.getfuncargvalue(fixturename)
+
+    if fixturename == 'atom':
+        myatoms = {obj}
+        mol = obj.molecule
+    else:
+        myatoms = set(obj.atoms)
+        mol = obj.atoms[0].molecule
+
+    assert len(obj.atoms_within(0.0*u.angstrom)) == 0
+
+    within5 = set(obj.atoms_within(5.0*u.angstrom))
+    within5_self = set(obj.atoms_within(5.0*u.angstrom, include_self=True))
+
+    assert myatoms.issubset(within5_self)
+    assert within5.union(myatoms) == within5_self
+
+    for atom in mol.atoms:
+        if atom in myatoms:
+            assert atom not in within5
+        elif atom.distance(obj) <= 5.0*u.angstrom:
+            assert atom in within5
+        else:
+            assert atom not in within5
+
+
+@pytest.mark.parametrize('fixturename', ['atom', 'residue', 'atomlist'])
+def test_residues_within(fixturename, request):
+    obj = request.getfuncargvalue(fixturename)
+
+    if fixturename == 'atom':
+        mol = obj.molecule
+    else:
+        mol = obj.atoms[0].molecule
+
+    assert len(obj.residues_within(0.0*u.angstrom)) == 0
+
+    within5 = set(obj.residues_within(5.0*u.angstrom))
+    within5_self = set(obj.residues_within(5.0*u.angstrom, include_self=True))
+
+    if fixturename == 'residue':
+        assert within5.union([obj]) == within5_self
+    else:
+        assert within5 == within5_self
+
+    for residue in mol.residues:
+        if residue == obj:
+            assert residue in within5_self
+            assert residue not in within5
+        elif residue.distance(obj) <= 5.0*u.angstrom:
+            assert residue in within5
+        else:
+            assert residue not in within5
```

Comment:
That's a good idea. I haven't really bothered to distinguish between unit tests and integration tests, but mocking is definitely a common Python testing pattern. We should at least think about about it.